pydantic/main.py

```diff
@@ -324,7 +340,8 @@ def construct(cls, **values: Any) -> 'BaseModel':
         Chances are you don't want to use this method directly.
         """
         m = cls.__new__(cls)
-        m.__setstate__(values)
+        fields_set = values.pop("__fields_set__", None)
```

Comment:
Single quotes always please unless the string contains single quotes.

But again, can we avoid doing `values.pop()`.

Reply:
I think I need to add the ability to pass in some sort of state from the calling instance. Can I change `construct` to take in a `fields_set` or should I add a new function like `construct_with_fields` or similar?

Reply:
`construct` can take more arguments, just `__init__` that should remain pure.